gcp_deepvariant_runner.py

```diff
@@ -166,6 +168,62 @@
 }}
 """
 
+# Following const values are used to automatically set the computational flags.
+_WGS_MODEL_IDENTIFIER = 'wgs_standard'
+_WES_MODEL_IDENTIFIER = 'wes_standard'
+_WES_LARGE_THR = 12 * 1024 * 1024 * 1024
+_WGS_LARGE_THR = 200 * 1024 * 1024 * 1024
+_WGS_SMALL_THR = 25 * 1024 * 1024 * 1024
+
+
+class BamCategories(enum.Enum):
+  """List of BAM categories that determine automatically assigned flags."""
+  WES_SMALL = 0
+  WES_LARGE = 1
+  WGS_SMALL = 2
+  WGS_MEDIUM = 3
+  WGS_LARGE = 4
+
+
+# Default optimal computational flag values, one per BAM category.
+_DEFAULT_FLAGS = {}
+_DEFAULT_FLAGS[BamCategories.WES_SMALL] = {
+    'make_examples_workers': 8,
+    'make_examples_cores_per_worker': 1,
+    'call_variants_workers': 1,
+    'call_variants_cores_per_worker': 4,
+    'gpu': True
+}
+_DEFAULT_FLAGS[BamCategories.WES_LARGE] = {
+    'make_examples_workers': 16,
+    'make_examples_cores_per_worker': 1,
+    'call_variants_workers': 2,
+    'call_variants_cores_per_worker': 4,
+    'gpu': True
+}
+_DEFAULT_FLAGS[BamCategories.WGS_SMALL] = {
+    'make_examples_workers': 32,
+    'make_examples_cores_per_worker': 1,
+    'call_variants_workers': 4,
+    'call_variants_cores_per_worker': 4,
+    'gpu': True
+}
+_DEFAULT_FLAGS[BamCategories.WGS_MEDIUM] = {
+    'make_examples_workers': 64,
+    'make_examples_cores_per_worker': 1,
+    'tpu': True
```

Comment:
I run extensive tests for make_examples but not call_variants since I was not sure whether or not GPU/TPU will be added to the list of our offerings. Value of call_variants flags are based on what we discussed offline.